### PR TITLE
Don't get nullable slot if property sub-pattern has error

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/59804

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs
Comment (333fred) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:190]: It feels like a better solution would be checking if the symbol of the pattern is a valid symbol kind for having a slot. In this case, the symbol is a NamedTypeSymbol, and thus it shouldn't have a slot. Bailing out if there are any errors feels a bit too broad.

**Review by 333fred — COMMENTED** [review]
Done review pass (commit 1)

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs
Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:190]: I had a specific check as you suggested in earlier iteration but felt that a broader check would be more defensive ;-)

Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:190]: I think the broader check is both too broad and incorrect, as there's nothing directly connecting the symbol being valid to the HasErrors flag.

Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:190]: With the broader check, only binding has to know what is correct. If the symbols are invalid, then it flags the error. So I don't understand what you mean by incorrect.

Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:190]: I mean that there is no guarantee that, just because HasErrors is false, that the pattern symbol can have a slot. HasErrors is a flag intended for suppressing later errors. Having it affect nullable analysis this way is not the intended purpose.

Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs]: As I explained above: bad symbol implies HasError, so the converse holds and can be depended on: !HasErrors implies valid symbol.
I don't think it's worth holding off the PR on a nitpick, so I made the change.

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs
Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs]: Thanks. I did think it was worth holding off approval for :).

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs
Comment (333fred) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs:191]:
```cs
subpattern.Member is BoundPropertySubpatternMember { Symbol.Kind: SymbolKind.Property or SymbolKind.Field } member
```

**Comment by 333fred** [comment]
Test failures look legitimate.

**Comment by 333fred** [comment]
Done review pass (commit 2)

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs
Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs]: That would break for extended property scenario. Will add test to illustrate

**Review by 333fred — APPROVED** [review]
LGTM (commit 4), assuming tests are passing.

**Pull request merged**
